### PR TITLE
Gate robo-DTT on DTS, DTT, and DTP all being yes.

### PR DESCRIPTION
**Pull request by ashercodeorg**
As suggested by @mehalshah in [Continuous DTTs](https://docs.google.com/document/d/1aJaL8AIT1a5PShaset8HojDbTnBVrvDEeHyATiOR9YQ).

### CODE REVIEW

File: bin/cron/deploy_to_test [hunk 1]
```diff
@@ -6,28 +6,24 @@ require 'cdo/developers_topic'
 require 'cdo/github'
 require 'cdo/only_one'
 
-DEVELOPERS = 'developers'.freeze
```
Comment (ashercodeorg) [inline bin/cron/deploy_to_test:9]: Unrelated cleanup (unused constant).

File: bin/cron/deploy_to_test [hunk 2]
```diff
@@ -6,28 +6,24 @@ require 'cdo/developers_topic'
...
-  unless DevelopersTopic.dtt?
-    ChatClient.message(
-      'infra-test',
-      'robo-DTT skipped (DTT was not yes)',
-      color: 'yellow'
-    )
-    return
-  end
+def early_exit_if_appropriate
+  no_branches = {DTP: DevelopersTopic.dtp?, DTT: DevelopersTopic.dtt?, DTS: DevelopersTopic.dts?}.
```
Comment (mehalshah) [inline bin/cron/deploy_to_test:13]: Should we block DTT if DTP is no? We usually leave it no during HOC.

Comment (ashercodeorg) [reply bin/cron/deploy_to_test:13]: Hmm...true. But I also think we tend to keep `test` clean during HOC. For now, I'm going to keep things as they are, as being conservative by default doesn't seem bad.
